### PR TITLE
Alex- Add Current Prompt modal to Weekly Summary and a copy prompt function

### PR DESCRIPTION
**Pull request by alexander-huerta**
# Description
This PR adds a modal that contains the most current AI prompt that users should be using with their text completion editor of choice, to edit their weekly summary Upon entering the modal, there is a button that allows the user to copy the prompt to their clipboard.

## Related PRS (if any):
This PR adds to progress made in PR [868](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/868) on the Weekly Summary component. 

Future plans for the Weekly Summary Component include:

next week
1. Add the Current Prompt as a value in the database 
 note: the `currentPrompt` variable on line 15 of `CurrentPromptModal.jsx` is a temporary placeholder
2. Allow owners to easily edit the Current Prompt value--  and have changes be reflected instantly for all users.

this month

3. Integrate AI into the editor and have the option to auto generate a summary based on a users tangible time logs from current week  

## Main changes explained:
- Created a modal  to display the current AI prompt 
- Added a Button to trigger this modal to open, and placed it in the weekly summary component, under the text form
  - Added a more info modal to this button (this was not in the requirements and may be unnecessary. I would appreciate opinions) 
- Added a copy button in the current prompt modal to copy the current prompt

## How to test:
1. Start backed-(HGNRest repository) from the development branch by running npm run build then npm  start
2. From this repository, check into current branch and run npm run start local
3. Log as any user (more than one ideally)
4. Follow the process you normally would to add a weekly summary 
5. Test the changes and functionality explained above 

## Screenshots or videos of changes:
btn

info_modal

### CODE REVIEW

**Review by 20chen-7 — DISMISSED** [review]
Code and function look great!
Screenshot 2023-06-07 at 12 23 30 PM
Screenshot 2023-06-07 at 12 23 24 PM

**Review by stevenpaalz — DISMISSED** [review]
This feature is working as intended. Nice work! Tested from admin, volunteer, and manager accounts.

Note: Please reach out to me regarding future functionality of integrating chat-gpt prompt directly into the modal via OpenAI API. I recently did this for another application I am working on.

Screenshot 2023-06-07 at 4 56 36 PM
Screenshot 2023-06-07 at 4 53 26 PM

**Review by nathanah — DISMISSED** [review]
Not sure if the button would be better above the text box, but it appears to work as intended.
example:
image

**Comment by one-community** [comment]
> Not sure if the button would be better above the text box, but it appears to work as intended.
> example:

Yes, that's where I envisioned this too. Please move it there and change the text to say "View and Copy Current AI Prompt"

**Review by eduhorta — DISMISSED** [review]
Hi, @alexander-huerta!
The feature is working as intended! Nice job! 👍👍
image

image

**Review by cvtqx — APPROVED** [review]
The new modal looks great and is copied to clipboard as described. Great job!

Button with a tooltip:

Screen Shot 2023-06-08 at 11 33 56 AM

Modal after 'copy' button is clicked:

Screen Shot 2023-06-08 at 11 34 29 AM

**Review by AuroraLovesWhales — DISMISSED** [review]
Hi @alexander-huerta! I have reviewed your PR and everything works as intended. Great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/f97f68ab-9825-4ad1-84b0-58e05aad96da

**Comment by AnishPandita** [comment]
Hi @alexander-huerta I tested the PR and its working as intended. Tested via different users as well. The prompt modal and copy functionally is great.
Screenshot 2023-06-08 at 7 28 33 PM
Screenshot 2023-06-08 at 7 28 47 PM

**Review by NidaZaki — DISMISSED** [review]
Works as intended! Nice work! The button and the modal including the tooltip
image
image

**Review by leonzh2k — DISMISSED** [review]
image
Info popup works:
image
Modal works;
image

Copying works:
image

**Comment by Lucas-E** [comment]
Thank you everyone, moving this to final review!

**Comment by alexander-huerta** [comment]
Thanks all! Final adjustments have been made. 
@one-community @Lucas-E 

1. Button moved above the form box and to the right
2. Button Text updated 

890_adjustments

**Comment by one-community** [comment]
> Thanks all! Final adjustments have been made. @one-community @Lucas-E
> 
> 1. Button moved above the form box and to the right
> 2. Button Text updated

Looks great, thank you!

**Review by bienzguoa — APPROVED** [review]
Works as intended! 
Screenshot 2023-06-10 at 2 23 32 AM
Screenshot 2023-06-10 at 2 26 29 AM
Screenshot 2023-06-10 at 2 26 54 AM

**Comment by KaixiangGU** [comment]
It works well! Good job!

**Review by EdwinLauM — APPROVED** [review]
Works as intended! Good job! 

Screen Shot 2023-06-10 at 12 48 46 PM

Screen Shot 2023-06-10 at 12 48 53 PM

File: src/components/WeeklySummary/WeeklySummary.css [hunk 1]
```diff
@@ -110,6 +110,11 @@
   border-color: #fff #dee2e6 #dee2e6;
 }
 
+.summary-instructions-row {
+  display: flex;
+  justify-content: space-between;
```
Comment (nathanah) [inline src/components/WeeklySummary/WeeklySummary.css:115]:
```suggestion
  justify-content: space-between;
  margin-right: 0;
```
Last thing so the button lines up with the editor, but everything looks good other than that.
image

**Review by nathanah — CHANGES_REQUESTED** [review]

**Review by nathanah — APPROVED** [review]

**Pull request merged**